pilot/pkg/leaderelection/k8sleaderelection/healthzadaptor.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sleaderelection
```

Comment:
Is there a reason to fork the k8s code? We already import the client-go repo. If you've made modifications we should point out the changes. Not sure if we could embed the struct to simplify this.

Reply:
Yeah so John added [prioritized leader election to the k8s API](https://github.com/kubernetes/kubernetes/pull/103442/files) but it is potentially far from merge so I forked it here. This change is needed for one revision to steal lock from a lower priority revision.